### PR TITLE
mypy fix population manager and population view

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: other/misc
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5342

### Changes and notes
mypy fix population manager and population view 

### Testing
tests pass

### CODE REVIEW

**Review by stevebachmeier — APPROVED** [review]

File: src/vivarium/framework/population/population_view.py [hunk 1]
```diff
@@ -453,10 +451,10 @@ def _ensure_coherent_initialization(
 
     @staticmethod
     def _update_column_and_ensure_dtype(
-        update: pd.Series,
-        existing: pd.Series,
+        update: pd.Series[Any],
```
[the remark this answers is not in the document]
Comment (rmudambi) [reply src/vivarium/framework/population/population_view.py:454]: Yes, I think we should not worry about TypeVar introduction for now, unless someone finds that introducing it makes type hinting easier

File: src/vivarium/framework/population/population_view.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ def __init__(
         self,
         manager: "PopulationManager",
         view_id: int,
-        columns: Union[List[str], Tuple[str]] = (),
-        query: Optional[str] = None,
+        columns: Sequence[str] = (),
+        query: str = "",
```
[the remark this answers is not in the document]
Comment (rmudambi) [reply src/vivarium/framework/population/population_view.py:47]: There were a few tests that were explicitly checking that the query was None. Once the tests were changed to check for "", they pass. The VPH tests all passed with this change.

**Review by albrja — APPROVED** [review]

**Review by patricktnast — APPROVED** [review]

File: src/vivarium/framework/population/manager.py [hunk 3]
```diff
@@ -169,26 +185,19 @@ def setup(self, builder):
...
         return "PopulationManager()"
 
     ###########################
     # Builder API and helpers #
     ###########################
 
-    def get_view(
-        self, columns: Union[List[str], Tuple[str]], query: Optional[str] = None
-    ) -> PopulationView:
+    def get_view(self, columns: Sequence[str], query: str = "") -> PopulationView:
```
[the remark this answers is not in the document]
Comment (rmudambi) [reply src/vivarium/framework/population/manager.py:200]: Per discussion we will convert strings to a list of length 1 in this case. Regarding the general question, we didn't arrive at a universal solution.

[the remark this answers is not in the document]
Comment (rmudambi) [reply src/vivarium/framework/population/manager.py:200]: This is being done in this PR: https://github.com/ihmeuw/vivarium/pull/485

**Pull request merged**